### DISCUSSION TITLE
Can't make "AI on the edge" work with Node-Red

### DISCUSSION THREAD

**Discussion by mimijojo** [jomjol/AI-on-the-edge-device#1586] [General] [answered]
AI on the edge (Release: v13.0.7 (Commit: 304b9e0)) up and running on my ESP32-CAM.  Cfr attached config.ini
NodeRed 1.3.4 active on my Raspberry Pi.  Cfr attached flows.json
Mosquitto MQTT is also active.  "mqtt in"-node tested successfully. "mosquitto_pub -t gazmeter -m "test" "command displays expected payload msg (="test").

Issue : For some reason, I'm unable to send counter information from AI on the edge to my NodeRed using MQTT.

Cfr attached AI on the edge log file

Any hint ?
[log AI on the edge.txt](https://github.com/jomjol/AI-on-the-edge-device/files/10226362/log.AI.on.the.edge.txt)
[flows.json.txt](https://github.com/jomjol/AI-on-the-edge-device/files/10226363/flows.json.txt)
[config.ini.txt](https://github.com/jomjol/AI-on-the-edge-device/files/10226364/config.ini.txt)

**Comment by caco3** [comment]
I do not have experience with NodeRed.

From AI on the edge it all looks ok ,and the messages get sent correctly to the broker.
You could use http://mqtt-explorer.com/ to check if the broker gets the messages.
If so, then the issue is somewhere on the NodeRed side, maybe you subscribed wrong or the wrong topics?

**Reply by mimijojo** [reply]
Thanks for your reply.
I tested my config with the http://mqtt-explorer.com/ tool.  This works as well as it works with my "mosquitto_pub -t gazmeter -m "test" " test.
So I'm still stuck.
 I guess probably some specific MQTT "AI on the edge" - MQTT NodeRed mismatch issue, but I can't find what exactly.
Any other hint ?

**Reply by caco3** [reply]
does the NodeRed receive your test messages?

maybe you better ask in a NodeRed forum.

**Reply by mimijojo** [reply]
My NodeRed displays correctly the payload test msgs I'm sending with both http://mqtt-explorer.com/ tool and mosquitto_pub.
So it sounds more like MQTT issue specific to AI on the edge.
I could indeed raise this issue on a NodeRed forum, but I'm not sure how to phrase it, so they don't push back to an AI on the edge issue.

**Reply by mimijojo** [reply]
Solution found !  Value to be configured in "Topic" field of Node-Red was : "**gazmeter/main/raw** ".  Information found in "AI on the edge" log file.

**Reply by caco3** [reply]
thats great!

You might how ever prefer to use `gazmeter/main/value`! The raw can contain `N` if the recognition fails!

**Comment by caco3** [comment, marked as answer]
Solved: https://github.com/jomjol/AI-on-the-edge-device/discussions/1586#discussioncomment-4403429
